Add Tailscale as Third Party Integration for Palette Edge

kreeuwijk opened the pull request:
## Describe the Change

This PR creates a new Third Party Integrations section for Palette Edge, with Tailscale being the first documented integration.

## Review Changes

💻 [Preview URL](https://deploy-preview-1878--docs-spectrocloud.netlify.app/)

🎫 No Jira Ticket

lennessyy commented:
Changing base to release-4-2 because I need to use the networking folder created in the 4.2 branch. I will manually port this to older branches once approved.

kreeuwijk commented:
@lennessyy I see you adjusted the procedure to now include the EdgeForge steps directly instead of referencing them. Doesn't that create the risk that we get a copy of the EdgeForge steps with every integration we document and that those steps become out of date when the EdgeForge workflow changes?

It also makes it harder to see what steps are specific for the integration vs what steps are generically copied from EdgeFlow; it makes it more likely that a user skips an important step because they think it's just an example copy from a generic step, but in reality it contains one important change that they overlooked.

lennessyy commented:
> @lennessyy I see you adjusted the procedure to now include the EdgeForge steps directly instead of referencing them. Doesn't that create the risk that we get a copy of the EdgeForge steps with every integration we document and that those steps become out of date when the EdgeForge workflow changes?
> 
> It also makes it harder to see what steps are specific for the integration vs what steps are generically copied from EdgeForge; it makes it more likely that a user skips an important step because they think it's just an example copy from a generic step, but in reality it contains one important change that they overlooked.

I primarily made the change because we were using step numbers like "step 5" in the earlier version, but those steps could easily change if we were to update EdgeForge and we would have no way of ensuring references to it would be updated, so I avoid referring to other how-tos like this. 

I did this before the holiday break with the intention of coming back to it and make further changes. Now that i'm looking at it, I do agree that this is too much duplication. Since we did say that users need to be familiar with EdgeForge workflow, I think we can afford to remove some of those repeated steps. I will make some edits to this current draft.

kreeuwijk commented:
@lennessyy I added a bit more information on why we do certain things, or to call out a specific change inside of a generic step. I also added a Troubleshooting section to explain how to deal with overlapping CIDRs.

lennessyy commented:
Since we released and 4.2 and upgraded to docusaurus 3, the merge conflicts have become too much so i will actually close this and start a new PR.

lennessyy commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
````diff
@@ -0,0 +1,249 @@
+
+2. Next, in the OS pack of your cluster profile, add the following:
+
+```yaml {14}
````
@kreeuwijk Hi, I think this is basically ready to merge. Just one more question - any reason to highlight line 14? It sounds like the user need to add the entire block to their profile

kreeuwijk replied:
@lennessyy oh sorry I didn't know what `{14}` did, I thought it was some layout hint or something so I just copied it.

kreeuwijk commented on docs/docs-content/palette-cli/commands/validator.md:
```diff
@@ -266,7 +266,7 @@ Each plugin may have its own set of failures. Resolving failures will depend on
 | vSphere | Folder missing or not accessible | The `spectro-templates` folder is missing or not accessible. Ensure the folder exists and the user account used by Palette or VerteX has read access to the folder. The `spectro-templates` folder is used by Palette and VerteX to download OVAs during the install. |
 
-Every 30 seconds, the Validator will continuously re-issue a validation and update the `ValidationResult` CR with the result of the validation. The validation results are hashed, and result events are only emitted if the result has changed. Once you resolve the failure, the Validator will update the `ValidationResult` CR with the new result.
+Every 30 seconds, the Validator will continouslly re-issue a validation and update the `ValidationResult` CR with the result of the validation. The validation results are hashed and result events are only emitted if the result has changed. Once you resolve the failure, the Validator will update the `ValidationResult` CR with the new result.
```
@lennessyy `validator.md` does not belong in this PR

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
````diff
@@ -0,0 +1,249 @@
+
+1. First, configure an IP Pool in Tailscale. We have found the following configuration works well to assign addresses in the new range to all nodes:
+
+```json
````
Let's clean up the white spacing on this code snippet.

lennessyy replied:
hmm, not sure why it's showing up like that - doesn't have the white space in my code editor

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
```diff
@@ -0,0 +1,249 @@
+
+7. Build the Edge device installation ISO and providers images. Afterward, push the provider images to an image registry. For more information, refer to [Build Edge Artifacts](../edgeforge-workflow/palette-canvos.md).
+
+8. Flash your external volume with the Edge installer ISO image. We recommend using [balena etcher](https://etcher.balena.io/) to flash your volume. 
```
Is there a specific reason for why we would recommend this specific tool versus just mentioning it as an example? In other Edge docs, we mention it as an option, but we stay away from tooling recommendations unless there is a good reason for it.

lennessyy replied:
```suggestion
8. Flash your external volume with the Edge installer ISO image. You can use [balena etcher](https://etcher.balena.io/) or any other tool of your choice to flash your volume. 
```

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
```diff
@@ -0,0 +1,249 @@
+
+1. Log in to [Tailscale console](https://login.tailscale.com/admin/machines). 
+
+2. In the **Machines** tab, your Edge device is displayed in the Machines list. You can SSH to your host from any device that is also connected to your Tailscale network.
```
```suggestion
2. In the **Machines** tab, your Edge device is displayed in the Machines list. You can SSH to your host from any device that is also connected to your Tailscale network. Check out the [Tailscale SSH](https://tailscale.com/kb/1193/tailscale-ssh) documentation page to learn more about SSH with Tailscale.
```
This could help newer users.

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
```diff
@@ -0,0 +1,249 @@
+
+### All Traffic Dropped for 100.64.0.0/10 CIDR Range 
+
+Tailscale uses the 100.64.0.0/10 range of IP addresses for your Tailnets. That means that by default, this address range (or parts of it) cannot be used for any of the following:
```
```suggestion
Tailscale uses the 100.64.0.0/10 range of IP addresses for your Tailnets. That means that by default, this address range, or parts of it, cannot be used for any of the following:
```
We try to avoid parentheses inside sentences that convey information. The sentences can be rewritten to a more formal tone by omitting the paratheses. Paratheses are also a bit distracting.

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
````diff
@@ -0,0 +1,249 @@
+          systemctl start tailscale-iptables-fix.timer
+```
+
+This will ensure Tailscale does not drop traffic for IP ranges that it doesn't own. This is due to a known bug in Tailscale. Even though we restricted the IP Pool, Tailscale still puts in `iptables` rule on every node that drops unknown traffic from any address in the entire 100.64.0.0/10 range. 
````
If we can link to the bug, that'd be awesome.

lennessyy replied:
Kevin said that he only managed to confirm it via email and this is not officially disclosed by tailscale

karl-cardenas-coding commented on docs/docs-content/clusters/edge/third-party-integrations/tailscale.md:
```diff
@@ -0,0 +1,249 @@
+
+  :::info
+
+  In the above `stages` block, you are using the device ID of your Edge device (read from `/sys/class/dmi/id/product_uuid`) as the hostname with which to register your device with Tailscale. For more information about how this ID is generated, refer to [Install Configurations](../edge-configuration/installer-reference.md#device-id-uid-parameters).
```
```suggestion
  In the above `stages` block, you are using the device ID of your Edge device that is read from the file **/sys/class/dmi/id/product_uuid**, as the hostname with which to register your device with Tailscale. For more information about how this ID is generated, refer to [Install Configurations](../edge-configuration/installer-reference.md#device-id-uid-parameters).
```

karl-cardenas-coding reviewed (approved):
Looks good, just some minor suggestions.
